src/main/kotlin/net/atos/zac/solr/SolrUtil.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Lifely
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+
+package net.atos.zac.solr
+
+private const val SOLR_ESCAPE = '\\'
+private const val SOLR_QUOTE = '\"'
+
+/**
+ * Produces a quoted Solr string, with properly encoded contents, from a raw Java string.
+ *
+ * @param value the raw unencoded string
+ * @return the encoded and quoted Solr string
+ */
+fun quoted(value: String) = SOLR_QUOTE.toString() + encoded(value) + SOLR_QUOTE
```

Comment:
Is this not easier?
```suggestion
fun quoted(value: String) = "$SOLR_QUOTE${encoded(value)}$SOLR_QUOTE"
```

Reply:
Or even as an extension function?

Reply:
Used your suggestion, thanks! Extension function is also nice yes. Maybe another day. :-)